bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVHandler.java

```diff
@@ -0,0 +1,575 @@
+                }
+            }
+        });
+    }
+
+    /**
+     * Handle channel commands
+     *
+     * @param channelUID - the channel, which received the command
+     * @param command - the actual command (could be instance of StringType,
+     *            DecimalType or OnOffType)
+     */
+    @SuppressWarnings("null")
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (command == RefreshType.REFRESH) {
+            // currently no channels to be refreshed
+            return;
+        }
+
+        try {
+            if (!isOnline() || command.toString().equalsIgnoreCase("PAIR")) {
+                logger.info("Device {0} is offline, try to (re-)connect", deviceName());
+                connectReceiver(); // reconnect to MR, throws an exception if this fails
+            }
+
+            logger.debug("Channel command for device {0}: {1} for channel {2}", thingConfig.getFriendlyName(),
+                    command.toString(), channelUID.getId().toString());
+            switch (channelUID.getId()) {
+                case CHANNEL_POWER: // toggle power
+                    logger.info("Toggle power, new state={0}", command.toString());
+                    control.sendKey("POWER");
+                    break;
+                case CHANNEL_CHANNEL:
+                    control.selectChannel(command.toString());
+                    break;
+                case CHANNEL_CHUP:
+                    control.sendKey("CHUP");
+                    updateState(CHANNEL_CHUP, OnOffType.OFF);
+                    break;
+                case CHANNEL_CHDOWN:
+                    control.sendKey("CHDOWN");
+                    updateState(CHANNEL_CHDOWN, OnOffType.OFF);
+                    break;
+                case CHANNEL_VOLUP:
+                    control.sendKey("VOLUP");
+                    updateState(CHANNEL_VOLUP, OnOffType.OFF);
+                    break;
+                case CHANNEL_VOLDOWN:
+                    control.sendKey("VOLDOWN");
+                    updateState(CHANNEL_VOLDOWN, OnOffType.OFF);
+                    break;
+                case CHANNEL_KEY:
+                    if (command.toString().equalsIgnoreCase("PAIR")) { // special key to re-pair receiver
+                        logger.info("PAIRing key received, reconnect device {0}", deviceName());
+                    } else {
+                        control.sendKey(command.toString());
+                    }
+                    break;
+                default:
+                    logger.fatal("Command for unknown channel {0}", channelUID.getAsString());
+            }
+        } catch (MagentaTVException | RuntimeException e) {
+            String errorMessage = MessageFormat.format("Channel operation failed: Command={0}, value={1}",
+                    command.toString(), channelUID.getId().toString());
+            logger.exception(e, errorMessage);
```

Comment:
change to logger .debug

Reply:
how to notify a user when the channel command was not executed? The user "clicks" a button, nothing happens, what's next if there is no info in the log?

Reply:
What could be the reason for not executing a command. Wouldn‘t it be better to set Thong OFFLINE and add message communication error?